### ISSUE TITLE
No keyboard or controller input when Xbox One controller connected via Bluetooth

### ISSUE THREAD

**Issue by someg33zer** [libretro/RetroArch#15200]
## Description

I’ve upgraded my RetroArch from 1.8 to 1.15 using the stable Ubuntu PPA and now when I start it with my Xbox One controller connected through Bluetooth, there’s no keyboard or controller input response in the main menu. Mouse works but nothing else. I’ve tried:

1. updating core info file, assets, controller profiles and databases
2. removing .config/retroarch/retroarch.cfg and starting from scratch

but still the same lack of input.

The jstest-gtk program shows the Xbox controller working fine.

If I disconnect the Bluetooth connection to the controller, the keyboard works again.

### Expected behavior

Xbox One controller connected through Bluetooth should be recognised and work as a joypad input and should not prevent keyboard input.

### Actual behavior

Xbox One controller connected through Bluetooth is recognised but does not work as joypad input and prevents keyboard input.

### Steps to reproduce the bug

1. Connect Xbox One controller through Bluetooth.
2. Run RetroArch.
3. Try using the joypad or keyboard.

### Version/Commit

- RetroArch: 1.15.0-r202303110511+666fbdcb38-167~ubuntu22.04.1

### Environment information

- OS: Ubuntu 22.04 Jammy
- Compiler: N/A

**Comment by ichijiyujin** [comment]
I made this account just now, this is my first time using and interacting with the GitHub community. My apologies if I unknowingly break some decorum.

OS: EndeavourOS (Arch Linux)
Version: RetroArch 1.15.0 from Arch repositories.

I think I'm experiencing the same issue with a generic bluetooth gamepad, but it seems whatever is going on is a lot more weird.
My gamepad is a Trust GXT 590, that Linux and Retroarch sees as Trust-GMP-04. It's a bluetooth device that also has a wireless dongle, through which the computer sees it as a generic Xbox 360 gamepad (this is relevant for later).

I initially found that for me the issue is inconsistent on start. Usually controls would be frozen on first start of Retroarch, but after closing it and restarting once or twice, both the gamepad and keyboard would work fine. After more testing I've found that what determines this inconsistent behavior is whether I touch any of the gamepad controls after connecting it, but before starting Retroarch.

It turns out, when connecting it via bluetooth, Linux initially reads the 4 axes of the 2 thumbsticks stuck on -32767. 
![Screenshot_20230421_135123](https://user-images.githubusercontent.com/131455785/233627222-8c844a9d-9c4e-46e8-a559-b68513234977.png)
Axes 1, 2, 3, 4 are the thumbsticks getting wrong values, 5 and 6 are the triggers that normally report -32767 while not pressed, and 32767 while pressed.

Any input on the gamepad will reset these values. So it seems to me that this incorrect reporting on the 4 axes is what freezes the keyboard and gamepad controls in Retroarch.

Except they don't seem to be frozen entirely. As it was mentioned earlier the mouse still works, and if I use it to start a game, I find that controls work perfectly fine in game. But if I pull up the Quick Menu while in game, the controls once again either work or not depending on whether I tried to input anything on the gamepad prior to launching the game. And if I used the gamepad while in game, and use the mouse to Close Content, the controls revert back to normal. Oh, and yes, F1 to open Quick Menu works at all times, and also the Select+Start combo for menu access.

A consistent way to reproduce the issue seems to be disconnecting and reconnecting the gamepad while in Retroarch. Although I noticed a difference, in that some buttons for menu control, like Ok and Back, or the Esc key are not frozen in these situations, just the navigational up/down/left/right.

Coming back to the dongle adapter I mentioned earlier, if I plug in the adapter, but still connect the gamepad in bluetooth mode, the issue doesn't present itself at all, even though I can confirm the 4 axes still report the wrong values upon connecting.

TL;DR: a workaround for me has been to input anything on the gamepad after connecting via bluetooth, but before opening Retroarch, but if you disconnect/reconnect while in Retroarch the issue will come back.

**Comment by ichijiyujin** [comment]
> I’m having a similar problem, but it seems to be hard to figure out when exactly it occurs. For me, it usually goes like this:
> 
> 1. launch Retroarch:
>    
>    * mouse, keyboard, and joypad work fine in the menu heavy_check_mark
> 2. launch a game:
>    
>    * keyboard and joypad work fine in the game heavy_check_mark
> 3. press the configured hotkey on the joypad or keyboard to bring up the RetroArch menu:
>    
>    * keyboard and joypad **no longer work in the menu, only the mouse does** x
>      
>      * From this point, I can still return to the game via the mouse. When the game is active again, the keyboard and joypad work fine in-game, and I can again bring up the RetroArch menu via the hotkey—after which the keyboard and joypad again won’t do anything in the menu.

I  can get the same behavior if my gamepad disconnects and then reconnects during gameplay. 
I can install version 1.9.13 of RetroArch through Lutris, and the issue doesn't exist at all for me in that version. So it's either 1.9.14 or 1.9.15 that introduced the problem it seems.

**Comment by jldelsolare** [comment]
I have the same issue; when I press the right trigger, the Xbox Series controller starts working again, but when pressed down, it's as if the trigger acts as the left analog stick for up and down. I still haven't been able to fix it.
Kubuntu 22.04
retroarch 1.16

**Comment by teeedubb** [comment]
I have the same issue and I'm actually surprised I found others with the issues.
Key board works great when no Xbox controller is connected. As soon as the Xbox S controller is connected via bluetooth the keyboard and controller do not work. If I turn the controller off the keyboard begins working.

retroarch 1.16.0 from the stable PPA
Ubuntu 22.04

The terminal output doesnt show anything between when I connect and disconnect the controller. Is there anyway I can supply more info?

**Comment by jldelsolare** [comment]
I have noticed that when activating the right trigger, it moves from top to bottom. It seems to be interpreting the trigger as if it were the left analog stick from top to bottom. In other words, it treats button 5 as if it were the analog stick. Here you can see the test I performed: https://www.youtube.com/watch?v=zukrixGRo0k

**Comment by nfp0** [comment]
The https://github.com/libretro/RetroArch/issues/15756 issue was recently fixed on the libdecor library here:
https://gitlab.freedesktop.org/libdecor/libdecor/-/merge_requests/132

I'm not sure if that was the source of this controller problem. So please test if this issue still remains after libdecor releases the fix.

**Comment by nfp0** [comment]
@Igetin That's right. My bad.
Since @cyrilpicard mentioned this issue on the locale issue, I thought they could've been related.

**Comment by cyrilpicard** [comment]
I am not sure that the bug I have encoutered is exactly the same as this one, but I can confirm that it does'nt occur anymore with the latest libdecor built from git master.

**Comment by hrzhu** [comment]
Have the same or similar problem on Kubuntu 22.04, with Xbox Series Controller.
I've tested Retroarch stable and nightly.

RA 1.16: When connected with Xbox Series Controller via BT, Controller doesn't work properly. keyboard and mouse input doesn't work either.

RA nightly(2023-10-28): When connected with Xbox Series Controller via BT, Controller doesn't work properly and keyboard doesn't work. Mouse works though.

**Comment by teeedubb** [comment]
I recently got a xbox wireless adapter (off aliexpress, pretty sure its a clone, but it works fine in linux and windows) and am using the xone fork at https://github.com/dlundqvist/xone for the driver.
RA now works without issue when using a xbox one controller.
I know this isnt an ideal solution, just putting it out there.

**Comment by nfp0** [comment]
From the last few posts it seems the libdecor issue is related to this somehow, after all.
The libdecor problem was also fixed on the RetroArch side here https://github.com/libretro/RetroArch/pull/15782 so people having problems please test with the latest nightly.

And make sure you're using the correct RetroArch driver for your controller. Most should work fine with udev, but I'm not sure if that's the correct one for the Xbox controller or not.

Also, make sure to test on a brand new config file. Your current one could have been corrupted by the libdecor bug.

**Comment by hrzhu** [comment]
The problem is still reproducible on latest nightly(2023-11-04). Changing input drive from the default x to udev doesn't fix it. The nightly appimage build uses separate config files from the one I installed from ppa.

I have also tried back-porting latest libdecor and sdl to 22.04. It doesn't fix the issue either.

I wonder if this issue effects distros other than Ubuntu.

**Comment by nfp0** [comment]
@hrzhu I was talking about the controller driver, not input driver. The default controller driver is udev.

Downgrading to a previous RetroArch version fixes the issue?

**Comment by hrzhu** [comment]
@nfp0 Sorry, I mistook input driver for controller driver. But that's probably irrelevant.

I just tried multiple old appimage builds. @Igetin was right. The last working version is 1.12(https://buildbot.libretro.com/stable/1.12.0/linux/x86_64/RetroArch.7z). It has been broken since 1.13.

**Comment by nfp0** [comment]
@hrzhu  I see. Then your issue seems to be unrelated with the libdecor one. :slightly_frowning_face:

**Comment by hrzhu** [comment]
I did a git biselect. The bug was introduced by @sonninnos  commit 1aa258faa0a12f17809524c267f290759c0b6407. Last working version is c4d2788c08ca4b478cbd00334c88691bfcc412f2.

**Comment by sonninnos** [comment]
I can't replicate in Windows with any controller. 8BitDo Pro 2 is the closest one since it uses regular bluetooth and is seen as Xbone controller.

There has been 2 essential related changes after that one, but apparently something is still missing:
https://github.com/libretro/RetroArch/pull/15478
https://github.com/libretro/RetroArch/pull/15740

**Comment by hrzhu** [comment]
I have 8bitdo Ultimate Controller. It just works fine. I have 8Bitdo Pro too but I haven't tested it. Maybe there is something special with Xbox Series X/S controller? The original reporter uses Xbox One controller.

**Comment by hrzhu** [comment]
Just tested on Windows. The bug is not present on Windows.

**Comment by cyrilpicard** [comment]
> From the last few posts it seems the libdecor issue is related to this somehow, after all. The libdecor problem was also fixed on the RetroArch side here #15782 so people having problems please test with the latest nightly.
> 
> And make sure you're using the correct RetroArch driver for your controller. Most should work fine with udev, but I'm not sure if that's the correct one for the Xbox controller or not.
> 
> Also, make sure to test on a brand new config file. Your current one could have been corrupted by the libdecor bug.

The fix works for me.

Test environment :
- Archlinux
- libdecor 0.2.0
- Xbox 360 wireless controller with official Microsoft wireless receiver
- Controller driver : udev


With retroarch 1.16.0.3 :
- Delete retroarch.cfg
- Start retroarch
- Controller and keyboard both works fine
- Close retroarch (retroarch.cfg is written with "," as decimal separator)
- Restart retroarch
- Neither controller or keyboard works
- Power off controller
- Keyboard is working again

With retroarch built from git : 
- All works as expected, retroarch.cfg is written with the correct decimal separator (".") and both controller and keyboard works fine.

**Comment by hrzhu** [comment]
There's still some other issue other than that locale one preventing the Xbox controller to work properly.

My testing procedure:
1. Compiling Retroarch git master.
1. Move my entire ~/.config/retroarch to other place.
1. Start retroarch. Connect my Xbox controller via BT. Keyboard works atm. Contoller doesn't work.
1. Download controller profiles. Controller doesn't work properly atm. Keyboard doesn't work. Controller doesn't respond to any input other than right trigger. When you press right trigger, cursor moves from top to bottom(or maybe the other way) nonstop. There's a chance you can make it stop but I don't know a reliable way to do it. ATM controller dpad works agains and keyboard works again. If you press trigger agains, cursor moves from top to bottom again. Keyboard doesn't work again.

I have tried an 8bitdo SN30 Pro+ as well, it works fine. It seems only the Xbox controller causing the problem. The last working version  is https://github.com/libretro/RetroArch/commit/c4d2788c08ca4b478cbd00334c88691bfcc412f2

**Comment by eomanis** [comment]
For me, a Bluetooth-connected Sony Dual Shock 4 gamepad appears to be affected; I haven't been able to play anything with RetroArch for some time now.

Any input method becomes completely ineffective. Mouse, keyboard and gamepad.

Here is a screen recording of the current clean-configuration user experience on Arch Linux as of Dec 2, 2023.

https://github.com/libretro/RetroArch/assets/5933155/dc95667e-1cf9-477b-8ee8-7d187a42b67f

**Comment by nfp0** [comment]
> 4\. Controller doesn't respond to any input other than right trigger. When you press right trigger, cursor moves from top to bottom(or maybe the other way) nonstop. There's a chance you can make it stop but I don't know a reliable way to do it. ATM controller dpad works agains and keyboard works again. If you press trigger agains, cursor moves from top to bottom again. Keyboard doesn't work again.

@hrzhu This totally feels like the controller is spamming a continuous input and preventing the keyboard from taking any action. Could still be related to the whole dots vs commas thing I mentioned in my previous post.

Please open and close RetroArch two times and check the decimal separator on your config file.

**Comment by nfp0** [comment]
@eomanis Great! Then test with the latest RetroArch nightly. This has been fixed there with this PR: https://github.com/libretro/RetroArch/pull/15782

Or if you prefer, you can instead install the git version of libdecor as a fix was also merged there [here](https://gitlab.freedesktop.org/libdecor/libdecor/-/merge_requests/132) but not released yet.

**Comment by hrzhu** [comment]
@nfp0 If you mean ~/.config/retroarch/retroarch.cfg, I don't have commas in that files. All my locales are en_US.

**Comment by nfp0** [comment]
@hrzhu Hmm, then that's a different issue. I assume you're not affected by the black background when creating a new config and reopening RetroArch as seen here: https://github.com/libretro/RetroArch/issues/15200#issuecomment-1837240352

**Comment by ichijiyujin** [comment]
It's a different issue than the config thing, and as far as I can tell it has to do with connecting via bluetooth on Linux. As I stated in my [earlier post,](https://github.com/libretro/RetroArch/issues/15200#issuecomment-1517703697) I have a cheap Trust controller, that can connect either via bluetooth, or via it's own dongle, in which case it presents as an Xbox 360 controller and it works fine 100% of the time. And I remember someone else saying that they solved their issue by using a wireless dongle instead of using bluetooth to connect.

When I first encountered the issue, back in April, I didn't have this config issue people are talking about, no black background problem. I believe I was using X11 at that time. Now I'm using Wayland. I haven't even opened Retroarch in months. Checking now, I did get the config issue, and fixed it by changing locale to en_US. But the original issue still persists as I described it previously. 

As I showed via a screenshot [earlier](https://github.com/libretro/RetroArch/issues/15200#issuecomment-1517703697), when initially connecting via bluetooth, Linux reads the controller as spamming both thumbsticks with -32767 on all axis, until the user gets to press any input on the controller, after which it shows normal values. This doesn't happen when connecting via a dongle, and this is probably what messes with Retroarch. 

I'm not a programmer, and I don't know much about all the config options Retroarch has. I described in my original post the best I can how the issue presents itself for me, how I can easily reproduce it and how I can mitigate the issue for myself. There's only so much I can help with unfortunately.

**Comment by eomanis** [comment]
> Or if you prefer, you can instead install the git version of libdecor as a fix was also merged there [here](https://gitlab.freedesktop.org/libdecor/libdecor/-/merge_requests/132) but not released yet.

Oh, now I get it. That library, `libdecor`, was messing with the application-global locale settings without being told to do any such thing. Rude.

On Arch, installing the [`libdecor-git` AUR package](https://aur.archlinux.org/packages/libdecor-git) fixed RetroArch for me.

Thanks for fixing my RetroArch installation 👌

**Comment by nfp0** [comment]
@eomanis  Don't thank me. I just found out what package was the culprit. :slightly_smiling_face: 

> On Arch, installing the [libdecor-git AUR package](https://aur.archlinux.org/packages/libdecor-git) fixed RetroArch for me.

Yes that's what I did, while the fix doesn't get a release.

**Comment by nfp0** [comment]
> It's a different issue than the config thing, and as far as I can tell it has to do with connecting via bluetooth on Linux.

@ichijiyujin Yes indeed, your case seems like a completely different issue. The fact that the controller is reporting wrong stick values makes it an issue external to RetroArch.
Maybe try observing the joystick bytes flowing from the controller with `btmon`. If it really is reporting the faulty values there before you input any button, then it's a controller issue and there's nothing much you can do about it other than your workaround.

**Comment by nfp0** [comment]
@hrzhu @Igetin I haven't read the whole thread, but have you guys tried observing what the controllers are reporting outside of RetroArch? Just to try and make some sense of what's happening.
If you're on KDE you can use the Settings app as seen [here](https://github.com/libretro/RetroArch/issues/15200#issuecomment-1517703697).

**Comment by hrzhu** [comment]
The controller looks good in KDE controller setting and works well in other emulator and steam.

I have a question. How do I know which file in ~/.config/retroarch/autoconfig is in use when I connect the controller?
When I start Retroarch 1.12(which is the last working version) and 1.13, the controller is recognised as Xbox One Elite Series 2 Controller. While in latest stable Retroach, it's recognised as Xbox One Elite Series 2. Apparently they use different configs.

**Comment by nfp0** [comment]
@hrzhu So RetroArch 1.12 works using the same autoconfig files? Or are you using the config files bundled with 1.12?

**Comment by hrzhu** [comment]
1.12 works after I updating controller profiles. So I assume they both use
the same latest auto-configs.

**Comment by hrzhu** [comment]
So this is an auto-config issue rather than Retroarch's.
To use Xbox Wireless Controller 2020 on Retroarch 1.16, you can try the following steps:
rm  `~/.config/retroarch/autoconfig` or make a backup first, and then update controller profiles in Retroarch.
I don't know why but seemly Retroarch use this [config](https://github.com/libretro/retroarch-joypad-autoconfig/blob/master/udev/Microsoft_X-Box_One_S_pad_v2.cfg).
Now create `Xbox Wireless Controller.cfg` under `~/.config/retroarch/autoconfig/udev` and fill the file with these contents:
```
input_driver = "udev"
input_device = "Xbox Wireless Controller"
input_vendor_id = "1118"
input_product_id = "2835"
input_b_btn = "0"
input_y_btn = "3"
input_select_btn = "10"
input_start_btn = "11"
input_up_btn = "h0up"
input_down_btn = "h0down"
input_left_btn = "h0left"
input_right_btn = "h0right"
input_a_btn = "1"
input_x_btn = "4"
input_l_btn = "6"
input_r_btn = "7"
input_l2_axis = "+5"
input_r2_axis = "+4"
input_l3_btn = "13"
input_r3_btn = "14"
input_l_x_plus_axis = "+0"
input_l_x_minus_axis = "-0"
input_l_y_plus_axis = "+1"
input_l_y_minus_axis = "-1"
input_r_x_plus_axis = "+2"
input_r_x_minus_axis = "-2"
input_r_y_plus_axis = "+3"
input_r_y_minus_axis = "-3"
input_gun_trigger_mbtn = "1
```
Now the controller should work properly in Retroarch.

Actually there is a config for this controller.
https://github.com/libretro/retroarch-joypad-autoconfig/blob/18b4579ff5ab8ada432dd1fce2b3fca82b075a09/udev/Microsoft_X-Box_Series_XS_pad.cfg#L6
But it use product id 2834 rather than 2835 and its mapping is not the same as the one I have.
This [config](https://github.com/libretro/retroarch-joypad-autoconfig/blob/35ebdc9023f256dba70e859a656503c67b80a31d/hid/Xbox%20Series%20XS%20(Bluetooth).cfg) does use 2835 product id but I don't know how to use hid driver.

**Comment by nfp0** [comment]
@hrzhu I'm not sure why that's happening but I've never used the AppImage version of RetroArch. I don't know how it's config works.

But regardless, if this is an autoconfig issue, maybe it's better to open an issue here on the autoconfig issue tracker:
https://github.com/libretro/retroarch-joypad-autoconfig

**Comment by jldelsolare** [comment]
Realize the process described, and the issues persisted. I pointed to nul all the entries, and the keyboard and mouse started working. I began manually assigning the buttons until I reached the x-axis, which is when the keyboard and mouse stopped working. I assigned null to the x-axis in the file, and everything started working again. I lack sufficient knowledge to proceed further. How could I provide more information on this matter?
My file ended up like this.
> ```
> input_l_x_plus_axis = "nul"
> input_l_x_minus_axis = "nul"
> input_l_y_plus_axis = "+1"
> input_l_y_minus_axis = "-1"
> input_r_x_plus_axis = "nul"
> input_r_x_minus_axis = "nul"
> input_r_y_plus_axis = "+3"
> input_r_y_minus_axis = "-3"
> ```

> So this is an auto-config issue rather than Retroarch's. To use Xbox Wireless Controller 2020 on Retroarch 1.16, you can try the following steps: rm `~/.config/retroarch/autoconfig` or make a backup first, and then update controller profiles in Retroarch. I don't know why but seemly Retroarch use this [config](https://github.com/libretro/retroarch-joypad-autoconfig/blob/master/udev/Microsoft_X-Box_One_S_pad_v2.cfg). Now create `Xbox Wireless Controller.cfg` under `~/.config/retroarch/autoconfig/udev` and fill the file with these contents:
> 
> ```
> input_driver = "udev"
> input_device = "Xbox Wireless Controller"
> input_vendor_id = "1118"
> input_product_id = "2835"
> input_b_btn = "0"
> input_y_btn = "3"
> input_select_btn = "10"
> input_start_btn = "11"
> input_up_btn = "h0up"
> input_down_btn = "h0down"
> input_left_btn = "h0left"
> input_right_btn = "h0right"
> input_a_btn = "1"
> input_x_btn = "4"
> input_l_btn = "6"
> input_r_btn = "7"
> input_l2_axis = "+5"
> input_r2_axis = "+4"
> input_l3_btn = "13"
> input_r3_btn = "14"
> input_l_x_plus_axis = "+0"
> input_l_x_minus_axis = "-0"
> input_l_y_plus_axis = "+1"
> input_l_y_minus_axis = "-1"
> input_r_x_plus_axis = "+2"
> input_r_x_minus_axis = "-2"
> input_r_y_plus_axis = "+3"
> input_r_y_minus_axis = "-3"
> input_menu_toggle_btn = "12"
> ```
> 
> Now the controller should work properly in Retroarch.
> 
> Actually there is a config for this controller. https://github.com/libretro/retroarch-joypad-autoconfig/blob/18b4579ff5ab8ada432dd1fce2b3fca82b075a09/udev/Microsoft_X-Box_Series_XS_pad.cfg#L6 But it use product id 2834 rather than 2835 and its mapping is not the same as the one I have. This [config](https://github.com/libretro/retroarch-joypad-autoconfig/blob/35ebdc9023f256dba70e859a656503c67b80a31d/hid/Xbox%20Series%20XS%20(Bluetooth).cfg) does use 2835 product id but I don't know how to use hid driver.

**Comment by nfp0** [comment]
@jldelsolare Sorry if this was asked before, but on the Game Controller menu on KDE's settings, do the inputs look normal to you?

**Comment by nfp0** [comment]
Are those -32767 the triggers? Are those supposed to look like that?
And is it normal that the X and Y axis be at absolute zero? Does the Xbox controller have some built-in dead zones? Do those show expected values when you move them?

**Comment by kragol** [comment]
I am experiencing a similar problem to some users in this thread. When the controller (XBOX 360 wired) is plugged in, keyboard and controller do not work in menu nor in game (mouse does work). (Note: no controller button has any effect, including right trigger).  When no controller is plugged in, the keyboard does work in menu, keeps working if a controller is plugged in but both stop working as soon as a game is launched. If I unplug all controllers while retroarch is running (and both keyboard and controller do nothing), it still doesn't work unless I restart xboxdrv.service (via systemd) at which point the keyboard starts working again in the menu.

After a bit of tweaking, I figured out the problem is related to the controller driver (udev by default). If I switch to another driver, then everything works fine after restarting retroarch. Any other driver will let the keyboard work again in the menu, however only the sdl2 driver allowed my controller to work too. 

My conclusion is there seems to be a problem with the udev controller driver, perhaps combined with the xboxdrv driver. Seems likely that the issue is not related to bluetooth.

Note : my config file does not have comma separators, so that seems unrelated too.

- Retroarch: 1.18.0
- OS: Arch Linux (up to date), kernel linux-zen 6.8.9 , xboxdrv 0.8.8
- Controller: Xbox 360 wired USB (using xboxdrv driver; no bluetooth)

**Comment by sonninnos** [comment]
Autoconf profiles are controller driver specific, so more than likely the profile binds makes it so that some navigation button/axis is constantly being pressed down, which is no wonder if the resting state is -max as seen above, and thus it overrides and ignores all other input.

So I'd try deleting the specific profile that gets detected, and try creating it from scratch for the particular driver.

**Comment by kragol** [comment]
Thanks @sonninnos . That is the problem indeed. The autoconf profile (udev version) for my controller (XboxGamepad(userspacedriver).cfg) has these problematic entries:
```
input_up_axis = "-5"
input_left_axis = "-4"
```
The problem is that axes +4 and +5 are triggers, so I am not sure what -4 and -5 mean in that case. If I remap these two (or comment them out in the autoconf file), then everything is OK (triggers are mapped to right and down in menus but that is not a big problem).

Conclusion is that the latest XboxGamepad(userspacedriver).cfg file is probably incorrect and should  be updated. Same probably goes with other Xbox gamepads profiles mentioned in the thread.

**Comment by nfp0** [comment]
@kragol Hmm, if that's the case, do you mind opening an issue on the [autoconfig](https://github.com/libretro/retroarch-joypad-autoconfig) repo?
